### PR TITLE
First POC to generate doc from code.

### PR DESCRIPTION
**Pull request by romain-growthbook**
This PR adds the first POC to generate doc content from the code.

One thing that could be enhanced is the doc generation hook-up. I couldn't find a notion of dependency/generated files directly in the `docusaurus` build pipeline.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "dev": "docusaurus start --port 3200 --no-open",
     "docusaurus": "docusaurus",
     "start": "docusaurus start --port 3200",
+    "gen": "ts-node -r tsconfig-paths/register ./scripts/gen-event-webhook-doc.ts",
```
Comment (romain-growthbook) [inline docs/package.json:9]: * Can this be generated each time a dependent file changes?
* Add result in the git repo/diff.

Comment (romain-growthbook) [reply docs/package.json:9]: Add pre-commit hook to regenerate.

**Comment by royalfig** [comment]
This is looking good!

It's possible (and probably preferable) to import the markdown without the plugin, `docusaurus-plugin-includes`.  

The docs make it seem like you could do this anywhere, but I got it to work like this:

- Add the generated md file to `/src/partials/` 
- Import the file in the Events file: `import EventWebhookList from "@site/src/partials/_event-webhook-list.md" and use it as a component: <EventWebhookList />

**Comment by romain-growthbook** [comment]
> This is looking good!
> 
> It's possible (and probably preferable) to import the markdown without the plugin, `docusaurus-plugin-includes`.
> 
> The docs make it seem like you could do this anywhere, but I got it to work like this:
> 
> * Add the generated md file to `/src/partials/`
> * Import the file in the Events file and use it as a component:
> 
> ```js
> import EventWebhookList from "@site/src/partials/_event-webhook-list.md"
> 
> <EventWebhookList />
> ```

Thanks, this is done now!

**Review by bryce-fitzsimons — APPROVED** [review]

**Pull request merged**